dataland-internal-storage/src/main/kotlin/db/migration/V19__MigrateEutaxonomyNonFinancialsExtendedDatapoints.kt

```diff
@@ -0,0 +1,117 @@
+        "transitionalShareInPercent",
+        "relativeShareInPercent",
+        "substantialContributionToClimateChangeMitigationInPercent",
+        "substantialContributionToClimateChangeAdaptationInPercent",
+        "substantialContributionToSustainableUseAndProtectionOfWaterAndMarineResourcesInPercent",
+        "substantialContributionToTransitionToACircularEconomyInPercent",
+        "substantialContributionToPollutionPreventionAndControlInPercent",
+        "substantialContributionToProtectionAndRestorationOfBiodiversityAndEcosystemsInPercent",
+        "absoluteShare",
+        "fiscalYearEnd",
+        "fiscalYearDeviation",
+        "alignedActivities",
+        "nonAlignedActivities",
+    )
+
+    /**
+     * Create a nested JSON object from a JSON object and a key.
+     * @param jsonObject JSON object
+     * @param key key corresponding to the JSON object
+     */
+    private fun createNestedJsonObject(jsonObject: JSONObject, key: String) {
+        val nestedJson = JSONObject()
+        nestedJson.put("value", jsonObject.getOrJavaNull(key))
+        jsonObject.put(key, nestedJson)
+    }
+
+    /**
+     * Check if the keys of a JSON object are relevant fields and, if so, created a nested JSON object.
+     * @param jsonObject JSON object
+     */
+    private fun checkForRelevantFieldsInJsonObjectKeys(jsonObject: JSONObject) {
+        jsonObject.keys().forEach {
+            if (it == "absoluteShare") {
+                val absoluteShare = jsonObject[it] as JSONObject
+                val amount = absoluteShare["amount"]
+                absoluteShare.remove("amount")
+                absoluteShare.put("value", amount)
+            } else if (it in relevantFields) {
+                createNestedJsonObject(jsonObject, it)
+            } else {
+                // Do nothing
+            }
+            checkRecursivelyForBaseDataPointsInJsonObject(jsonObject, it)
+        }
+    }
+
+    /**
+     * Check recursively for BaseDataPoints in a JSON array.
+     * @param jsonArray JSON array
+     */
+    private fun checkRecursivelyForBaseDataPointsInJsonArray(jsonArray: JSONArray) {
+        for (i in 0 until jsonArray.length()) {
+            val element = jsonArray[i]
+            if (element != null && element is JSONObject) {
+                checkForRelevantFieldsInJsonObjectKeys(element)
+            }
+        }
+    }
+
+    /**
+     * Check recursively for BaseDataPoints in a JSON object.
+     * @param jsonObject JSON object
+     * @param key key corresponding to the JSON object
+     */
+    private fun checkRecursivelyForBaseDataPointsInJsonObject(jsonObject: JSONObject, key: String) {
+        val obj = jsonObject.getOrJavaNull(key)
+        if (obj !== null && obj is JSONObject) {
+            checkForRelevantFieldsInJsonObjectKeys(obj)
+        } else if (obj != null && obj is JSONArray) {
+            checkRecursivelyForBaseDataPointsInJsonArray(obj)
+        } else {
+            // Do nothing
```

Comment:
remove

Reply:
Is marked as a change because I did not merge main into branch earlier